packages/tracking/tests/createTrackingFunction.test.ts

```diff
@@ -0,0 +1,104 @@
+import { beforeEach, test, vi } from "vitest";
+import { createTrackingFunction } from "../src/createTrackingFunction.ts";
+
+const fetchMock = vi.fn();
+globalThis.fetch = fetchMock;
+
+beforeEach(() => {
+    vi.clearAllMocks(); // clears call history but keeps mock implementations
+});
+const mockBaseProps = { baseProp: "base" };
+vi.mock("../src/properties.ts", () => ({
+    getBaseProperties: () => mockBaseProps
+}));
+
+test("when called, sends a request to the default endpoint with merged properties", async ({ expect }) => {
```

Comment:
Wondering, have you try running the test concurrently? I know there's not many but when possible, it seems to be a good idea to always aim at running test concurrently. It also give you a hint sometimes when it's not possible that something is wrong with the code (sometimes it's not just possible though, which is also fine).